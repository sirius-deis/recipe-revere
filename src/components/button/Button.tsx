import { FC, PropsWithChildren } from "react";
import styles from "./Button.module.css";

interface ButtonProps {
  size?: "lg" | "sm" | "md";
  bg?: "main" | "action" | "icon";
  rounded?: true | false;
  inlineStyles?: {};
  clickHandler?: () => void;
}

const Button: FC<PropsWithChildren & ButtonProps> = ({
  children,
  size,
  bg,
  inlineStyles,
  rounded,
  clickHandler = () => {},
}): JSX.Element => {
  return (
    <button
      className={`${styles.btn} ${
        size ? `${styles[`btn-${size}`]}` : "btn-lg"
      } ${bg ? `${styles[`btn-${bg}`]}` : "btn-main"} ${
        rounded ? `${styles["btn-rounded"]}` : ""
      } `}
      style={inlineStyles}
      onClick={clickHandler}
    >
      {children}
    </button>
  );
};

export default Button;
