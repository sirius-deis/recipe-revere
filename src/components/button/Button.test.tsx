import { render, screen, fireEvent } from "@testing-library/react";
import Button from "./Button";

describe("Button component", () => {
  it("should match snapshot", () => {
    const { container } = render(<Button>Get Started</Button>);
    expect(container).toMatchSnapshot();
  });
  it("should render component with proper text", () => {
    render(<Button>Get Started</Button>);
    expect(screen.getByText("Get Started")).toBeInTheDocument();
    expect(screen.getByRole("button")).toHaveTextContent("Get Started");
  });
  it("should render component with html inside it", () => {
    render(
      <Button>
        <h2>Get Started</h2>
      </Button>
    );
    expect(screen.getByRole("button")).toContainHTML(
      '<button class="btn btn-lg btn-main"><h2>Get Started</h2></button>'
    );
  });
  it("should fire provided function on click event", () => {
    const fn = jest.fn();
    render(<Button clickHandler={fn}>Click me</Button>);
    fireEvent.click(screen.getByRole("button"));
    expect(fn).toHaveBeenCalled();
  });
  it("should check if default classes are provided correctly", () => {
    render(<Button>Click me</Button>);
    expect(screen.getByRole("button")).toHaveClass("btn");
    expect(screen.getByRole("button")).toHaveClass("btn-lg");
    expect(screen.getByRole("button")).toHaveClass("btn-main");
    expect(screen.getByRole("button")).not.toHaveClass("btn-rounded");
  });
});
