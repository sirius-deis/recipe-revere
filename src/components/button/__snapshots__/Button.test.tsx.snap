// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<div>
  <button
    class="btn btn-lg btn-main"
  >
    Get Started
  </button>
</div>
`;
